contrib/hyperscan/matching/input_matchers/source/matcher.cc

```diff
@@ -34,8 +34,9 @@ bool Bound::operator<(const Bound& other) const {
 
 Matcher::Matcher(const std::vector<const char*>& expressions,
                  const std::vector<unsigned int>& flags, const std::vector<unsigned int>& ids,
-                 ThreadLocal::SlotAllocator& tls, bool report_start_of_matching)
-    : tls_(ThreadLocal::TypedSlot<ScratchThreadLocal>::makeUnique(tls)) {
+                 Event::Dispatcher& dispatcher, ThreadLocal::SlotAllocator& tls,
+                 bool report_start_of_matching)
+    : dispatcher_(dispatcher), tls_(ThreadLocal::TypedSlot<ScratchThreadLocal>::makeUnique(tls)) {
```

Comment:
nit: can you rename this local and param to main_thread_dispatcher ?

Reply:
Ok, changed.